python_modules/dagster/dagster/_core/definitions/automation_condition_sensor_definition.py

```diff
@@ -46,7 +48,14 @@ def _evaluate(sensor_def: "AutomationConditionSensorDefinition", context: Sensor
         emit_backfills=sensor_def.emit_backfills,
         default_condition=sensor_def.default_condition,
         logger=context.log,
-    ).evaluate()
+    )
+    check.invariant(
+        evaluation_context.total_keys <= MAX_ENTITIES,
+        f'AutomationConditionSensorDefintion "{sensor_def.name}" targets {evaluation_context.total_keys} '
+        f"assets or checks, which is more than the limit of {MAX_ENTITIES}. Either set `use_user_code_server` to `False`, "
+        "or split this sensor up into smaller selections.",
```

Comment:
maybe more detail on what splitting in to smaller selections means exactly or how to do it